Bwt/SuffixArray/SuffixArray.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace BWT;
+/// <summary>
+/// Suffix Array class
+/// </summary>
+public class SuffixArray
+{
+    public readonly int[] SufArray;
+    private readonly Suffix StartString; 
+
+    /// <summary>
+    /// Constructor of <see cref="SuffixArray"/>
+    /// </summary>
+    public SuffixArray(String s)
+    {
+
+        var n = s.Length;
+        var suffix = new Suffix[n];
+        
+        for (var i = 0; i < n; i++)
+        {
+            suffix[i] = new Suffix(i, s[i] - '$', 0);
+        }
+
+        for (var i = 0; i < n; i++)
+            suffix[i].Next = i + 1 < n ? suffix[i + 1].Rank : -1;
```

Comment:
```suggestion
        for (var i = 0; i < n; i++)
        {
            suffix[i].Next = i + 1 < n ? suffix[i + 1].Rank : -1;
        }
```
Во избежание нелепых ошибок вида
```csharp
for (var i = 0; i < n; i++)
    Console.WriteLine($"{n}");
    suffix[i].Next = i + 1 < n ? suffix[i + 1].Rank : -1;
```